Cargo.toml

```diff
@@ -23,6 +23,7 @@ regex = { version = "1.5.5", default-features = false, features = ["std", "unico
 aho-corasick = "0.7"
 tantivy-fst = "0.4.0"
 memmap2 = { version = "0.5.3", optional = true }
+normpath = "1.0.0"
```

Comment:
I'd rather not introduce a dependency for this. Can you remove it and stick to std's canonicalize?

Reply:
> Can you remove it and stick to std's canonicalize?

Unfortunately this is not possible currently as the [PR](https://github.com/rust-lang/rust/pull/86447) attempting to fix the problem was closed, but in the future we can switch to [`std::path::absolute`](https://doc.rust-lang.org/std/path/fn.absolute.html) once it is stabilized.

IMHO `normpath` is quite lightweight, and its biggest dependency is `windows-sys = { version = "0.45", features = ["Win32_Storage_FileSystem"] }` (not the macro-heavy `windows` crate), which only adds some reasonable amount of compilation time and little extra size to the release build. Also Tantivy has already included dependencies of `windows` 0.39 and `windows-sys` 0.42.

<details>
<summary>cargo-bloat output</summary>

```
    Analyzing target\release\bloat-normpath.exe

 File  .text     Size    Crate Name
 1.9%   2.6%   3.2KiB      std rustc_demangle::demangle
 1.7%   2.4%   3.0KiB      std <rustc_demangle::legacy::Demangle as core::fmt::Display>::fmt
 1.4%   1.9%   2.4KiB normpath normpath::normalize::normalize_virtually
 1.3%   1.7%   2.2KiB      std core::str::pattern::impl$28::is_contained_in
 1.1%   1.5%   1.9KiB      std <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
 1.0%   1.4%   1.8KiB      std rustc_demangle::v0::Printer::print_path
 1.0%   1.4%   1.7KiB      std std::sys_common::once::generic::Once::call
 1.0%   1.4%   1.7KiB      std std::sys::windows::path::maybe_verbatim
 1.0%   1.4%   1.7KiB      std rustc_demangle::v0::Printer::print_const
 1.0%   1.3%   1.7KiB      std core::str::count::do_count_chars
56.7%  77.7%  97.9KiB          And 443 smaller methods. Use -n N to show more.
73.0% 100.0% 126.0KiB          .text section size, the file size is 172.5KiB
```
</details>

Reply:
@gyk I won't merge this with the extra dependency. Isn't this PR useful without the canonicalization?

Reply:
No, with `std::fs::canonicalize` the index still cannot be put on special drives. You might want to consider:

1. Make normpath a Windows-only dependency, or
2. Call `std::fs::canonicalize` like before, but in case it returns this kind of error but the path does exist, just use the original `directory_path` as a fallback.

Reply:
ah ok so canonicalize return an error for that kind of path. I see. I'm ok with solution 1 or solution 2. You can decide.

Reply:
OK, I'll update the code later.

Reply:
@fulmicoton Updated using approach 2 above. PTAL.